pkg/ceph/client/osd.go

```diff
@@ -65,6 +65,33 @@ type OSDDump struct {
 	} `json:"osds"`
 }
 
+// StatusByID returns status and inCluster states for given OSD id
+func (dump *OSDDump) StatusByID(id int64) (int64, int64, error) {
+	for _, d := range dump.OSDs {
+
+		i, err := d.OSD.Int64()
+		if err != nil {
+			return 0, 0, err
+		}
+
+		if id == i {
+			inCluster, err := d.In.Int64()
+			if err != nil {
+				return 0, 0, err
+			}
+
+			status, err := d.Up.Int64()
+			if err != nil {
+				return 0, 0, err
+			}
+
+			return status, inCluster, nil
+		}
+	}
+
+	return 0, 0, fmt.Errorf("not found OSD id%d in OSDDump", id)
```

Comment:
in general, we have normally logged OSD ID's as something like `"osd %d"` or `"osd.%d"`...i would consider logging in that format as well to make greps for a specific ID across the entire log more fruitful.